lib/portage/util/socks5.py

```diff
@@ -74,10 +81,15 @@ def stop(self) -> Union[None, asyncio.Future]:
         if self._proc is not None:
             self._proc.terminate()
             if loop is None:
-                asyncio.run(self._proc.wait())
+                # In this case spawn internals would have used
+                # portage's global loop when attaching a waiter to
+                # self._proc, so we are obligated to use that.
+                global_event_loop().run_until_complete(self._proc.wait())
```

Comment:
We can actually use the real `asyncio.get_event_loop()` in cases like this without triggering a DeprecationWarning, since portage's _safe_loop will have set the real asyncio global event loop like this:
```python
            except RuntimeError:
                _loop = _real_asyncio.new_event_loop()
                _real_asyncio.set_event_loop(_loop)
```